tests/report_downstream_test_diffs.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import json
+import os
+import re
+import subprocess
+import tempfile
+from collections import namedtuple
+
+TestCase = namedtuple("TestCase", ["api", "file", "test"])
+
+
+def pytest_test_cases():
+    p = subprocess.run(
+        ["pytest", "--browser", "chromium", "--collect-only", "-q"],
+        stdout=subprocess.PIPE,
+    )
+    regex = (
+        r"tests/(?P<api>a?sync)/test_(?P<file>.*)\.py::test_(?P<test>.*)\[chromium\]"
+    )
+    matches = re.finditer(regex, p.stdout.decode(), re.MULTILINE)
+    for match in matches:
+        yield TestCase(
+            match.group("api"), match.group("file"), match.group("test"),
+        )
+
+
+def jest_test_cases(playwright_js_path):
+    env = os.environ.copy()
+    env["REPORT_ONLY"] = "true"
+    with tempfile.NamedTemporaryFile() as results:
+        subprocess.run(
+            ["npx", "jest", "--json", "--outputFile", results.name],
+            env=env,
+            cwd=playwright_js_path,
+            stderr=subprocess.DEVNULL,
+        )
+
+        for test_suite in json.load(results)["testResults"]:
+            regex = r"(.*/)?(?P<file>[^/]+)\.spec\.[jt]s$"
+            file = re.match(regex, test_suite["name"]).group("file")
+            for assertion in test_suite["assertionResults"]:
+                yield TestCase("sync", normalized(file), normalized(assertion["title"]))
+                yield TestCase(
+                    "async", normalized(file), normalized(assertion["title"])
+                )
+
+
+def normalized(original):
+    return re.sub(r"[^a-z_]", "_", original, flags=re.IGNORECASE)
```

Comment:
@pavelfeldman This is the normalization bit. We get some odd names because of it:

For example:

```
async/test_eval_on_selector.py::test_should_support____syntax
async/test_page_evaluate.py::test_should_return___
async/test_page_evaluate.py::test_should_transfer____Mb_of_data_from_page_to_node_js
```

Reply:
Makes sense. How about we bring numbers back, lowercase all and squash `([_]+)` into `_`?

Reply:
Brought numbers back, squashed repeated underscores, and stripped trailing underscores.